diff --git a/studiocomponents/build/components/common/input.lua b/studiocomponents/build/components/common/input.lua
index bccc406..7404907 100644
--- a/studiocomponents/build/components/common/input.lua
+++ b/studiocomponents/build/components/common/input.lua
@@ -1,4 +1,5 @@
 local CoreGui = game:GetService("CoreGui")
+local PluginDebugService = game:GetService("PluginDebugService")
 local StarterGui = game:GetService("StarterGui")
 
 local fusion = require(script.Parent.Parent.Parent.Parent.fusion)
@@ -25,7 +26,7 @@ local baseButton = require(script.Parent.Parent.base.button)
 local baseTextbox = require(script.Parent.Parent.base.textBox)
 local baseIcon = require(script.Parent.Parent.base.icon)
 
-local plugin = script:FindFirstAncestorWhichIsA("Plugin")
+local plugin = script:FindFirstAncestorWhichIsA("Plugin") or PluginDebugService:FindFirstChild("plugin") :: Plugin
 local pluginMouse = plugin and plugin:GetMouse() or nil
 
 type props = {
